dig.go

```diff
@@ -265,12 +256,61 @@ var _graphTmpl = template.Must(
 // Visualize parses the graph in Container c into DOT format and writes it to
 // io.Writer w.
 func Visualize(c *Container, w io.Writer, opts ...VisualizeOption) error {
-	if err := _graphTmpl.Execute(w, c.dg); err != nil {
+	dg := c.createGraph()
+
+	if err := _graphTmpl.Execute(w, dg); err != nil {
 		return fmt.Errorf("error executing template: %v", err)
 	}
 	return nil
 }
 
+// VisualizeError parses the container and errors included in err into a DOT
+// format graph and writes it to io.Writer w.
+func VisualizeError(err error, w io.Writer, opts ...VisualizeOption) error {
+	ep, ok := err.(errEntryPoint)
+	if !ok {
+		return fmt.Errorf("No graph included in error: %v", err)
+	}
+
+	var errors []errVisualizer
+
+	for {
+		if ev, ok := err.(errVisualizer); ok {
+			errors = append(errors, ev)
+		}
+		e, ok := err.(causer)
+		if !ok {
+			break
+		}
+		err = e.cause()
+	}
+
+	if len(errors) == 0 {
+		return fmt.Errorf("cannot visualize error: %v", err)
+	}
+
+	dg := ep.container().createGraph()
+	for i := len(errors) - 1; i >= 0; i-- {
```

Comment:
Any particular reason for the reverse iteration?

Reply:
Yes. As we unwrap the errors the outermost error becomes the first element in `errors`, and the innermost error becomes the last element. We want to call `updateGraph` on the innermost error first since it is the root cause, so we iterate in reverse.

Reply:
We should add a comment here about this.

Reply:
Cool. Might be worth adding a simple in-line comment here so that this behavior isn't accidentally changed in the future.